src/fox-news/fox-news.js

```diff
@@ -1,146 +1,168 @@
+  return hours
+}
 
-const discoverAllHomepageArticles = async page => {
-  await page.goto(FOX_NEWS_URL);
-  await allDOMContent(page, "h2.title a")
-    .then(articles => {
-      store.dispatch(
-        addArticles(
-          articles.map(article => ({
-            ...article,
-            title: article.innerHTML.trim()
-          }))
-        )
-      );
-    })
-    .catch(console.error);
-};
+const parseRelativeDate = relativeDate => {
+  const date = new Date()
+  const hoursAgo = parseHoursAgo(relativeDate)
+  assert(hoursAgo, 'Cannot parse relative date')
+  if (hoursAgo) {
+    date.setHours(date.getHours() - Number(hoursAgo))
+  }
+  return date.toString()
+}
 
-const articlePageContent = async page => {
-  const embeddedTwitterMedia = await allDOMContent(
-    page,
-    ".article-body .embed-media.twitter"
-  );
-  const authors = await allDOMContent(
-    page,
-    '.author-byline a[href^="/person/"]'
-  );
-  const videoContainer = await page.$(".video-container");
-  const ps = await articleParagraphs(page);
-  const timestamp = await page
-    .$eval("time", time => ({
-      scrapeDate: Date(),
-      relativeDate: time.innerHTML.trim()
-    }))
-    .catch(() => ({ scrapeDate: Date(), relativeDate: null }));
-  const images = await page.$$eval(".image-ct.inline", imageSections =>
-    Promise.all(
-      imageSections.map(async section => {
-        const img = section.querySelector("img");
-        const p = section.querySelector(".caption p");
-        return {
-          imageSrc: img ? img.src : null,
-          caption: p
-            ? p.innerHTML
-                .replace("\n", "")
-                .replace(/ */, " ")
-                .trim()
-            : null
-        };
-      })
-    )
-  );
+const articleContent = async page => {
+  const title = await page.$eval(
+    'header .headline',
+    headline => headline.innerText
+  )
+  const authors = await page.$$eval(
+    '.author-byline a[href^="/person/"]',
+    links =>
+      links.map(l => ({
+        href: l.href,
+        name: l.innerText,
+      }))
+  )
+  const content = await articleParagraphs(page)
+  const relativePublicationDate = await page.$eval('time', time =>
+    time.innerHTML.trim()
+  )
   return {
+    href: page.url(),
     authors,
-    content: ps.map(p => p.innerHTML).join("\n"),
-    hasVideo: Boolean(videoContainer),
-    images,
-    timestamp,
-    twitterContent:
-      embeddedTwitterMedia
-        .map(
-          content =>
-            // TODO Maybe clean this up. It looks like we can pull the twitter id
-            // which maps directly to a url like
-            // `twitter.com/literally-a-random-user-name/status/<twitter-id>`
-            `<div class=".embed-media-twitter">${content.innerHTML}</div>`
-        )
-        .join("") || null
-  };
-};
+    title,
+    content: content.join('\n'),
+    publicationDate: parseRelativeDate(relativePublicationDate),
+  }
+}
+
+const discover = async page => {
```

Comment:
I think this discovering strategy works out pretty well. It make implementation come down to configuration. I don't think I have it perfect, but the essential workflow is
- Collect every `a[href]` that's relevant to the newsSource
- Figure out the content structure of the website according to `how/the/hrefs/are/pathed`
- Bucket the hrefs such that they're routed either to more discovery or to article content collection